## Parte I: Cargar y observar como esta distribuido el dataset

In [1]:
# Loading data
# ------------

import pandas as pd

df =  pd.read_csv('../data/most_used_beauty_cosmetics_products_extended.csv')
df.head()

Product_Name              Brand        Category Usage_Frequency  \
0   Ultra Face Mask     Drunk Elephant           Blush          Weekly   
1    Ultra Lipstick      Laura Mercier  Makeup Remover      Occasional   
2       Ultra Serum     Natasha Denona     Highlighter           Daily   
3      Divine Serum        Ilia Beauty       Face Mask      Occasional   
4  Super Foundation  Charlotte Tilbury     Highlighter      Occasional   

   Price_USD  Rating  Number_of_Reviews Product_Size  Skin_Type Gender_Target  \
0      67.85     1.4                686         30ml  Sensitive        Female   
1     116.43     4.2               5483        250ml        Dry        Unisex   
2      90.84     1.6               5039        100ml  Sensitive          Male   
3      55.17     3.2               6202        250ml     Normal          Male   
4     140.56     1.7                297        100ml       Oily        Female   

  Packaging_Type Main_Ingredient  Cruelty_Free Country_of_Origin  
0           Tube         Retinol         False         Australia  
1         Bottle     Shea Butter         False                UK  
2        Compact       Aloe Vera          True             Italy  
3           Tube        Glycerin          True       South Korea  
4        Compact        Glycerin         False           Germany

In [2]:
# Basic inspection
# ----------------
# Descriptive statistics
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15000 entries, 0 to 14999
Data columns (total 14 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Product_Name       15000 non-null  object 
 1   Brand              15000 non-null  object 
 2   Category           15000 non-null  object 
 3   Usage_Frequency    15000 non-null  object 
 4   Price_USD          15000 non-null  float64
 5   Rating             15000 non-null  float64
 6   Number_of_Reviews  15000 non-null  int64  
 7   Product_Size       15000 non-null  object 
 8   Skin_Type          15000 non-null  object 
 9   Gender_Target      15000 non-null  object 
 10  Packaging_Type     15000 non-null  object 
 11  Main_Ingredient    15000 non-null  object 
 12  Cruelty_Free       15000 non-null  bool   
 13  Country_of_Origin  15000 non-null  object 
dtypes: bool(1), float64(2), int64(1), object(10)
memory usage: 1.5+ MB


- El total de las filas existentes son 15000 y estan enumeradas de 0 a 14999El total de columnas son 14
- No hay datos faltantes en las diferentes columnas (15000 non-null)
- Hay 15000 productos en total
- Los datos existentes para cada columna son:
    - 0   Product_Name -> String
    - 1   Brand        -> String 
    - 2   Category     -> String  
    - 3   Usage_Frequency -> String 
    - 4   Price_USD    -> float
    - 5   Rating       -> float64
    - 6   Number_of_Reviews ->  int
    - 7   Product_Size  -> String
    - 8   Skin_Type     -> String
    - 9   Gender_Target -> String
    - 10  Packaging_Type -> String
    - 11  Main_Ingredient  -> String
    - 12  Cruelty_Free  ->  bool   
    - 13  Country_of_Origin  -> String

  

In [28]:
df.describe()

Price_USD        Rating  Number_of_Reviews
count  15000.000000  15000.000000       15000.000000
mean      80.134108      3.002327        5014.231333
std       40.402983      1.168029        2855.665464
min       10.000000      1.000000          52.000000
25%       45.480000      2.000000        2562.000000
50%       80.040000      3.000000        5002.000000
75%      114.760000      4.000000        7497.000000
max      149.990000      5.000000       10000.000000

- **Price_USD:**
    - **Promedio:** 80.13 USD, Los productos posee en promedio un precio de 80.13 USD
    - **Rango:** De 10 USD a 114.76 USD, hay productos con el precio de 10 y productos mas costoso de 149.99
    - **Desviación Estandar:** 40.40 USD respecto al promedio, sugiere que es una variablidad media a alta, ya que la desviacion estandar representa el 50.41 % del promedio, sugiriendo que hay productos cuya variabilidad de precios esta alejado del promedio, duplicando incluso el valor del precio promedio.
    - **Quartiles:**
        - El 25% de los productos cuesta 45.48 USD o menos, lo que quiere decir que los productos mas economico esta por debajo de este precio
        - El 50% de los productos posee un precio de 80.04 USD, este hace referencia a la mediana, indicando que la mitad de los precios de los productos esta por debajo DE 80.04 USD y la otra por encima
        - El 75% de los productos posee un precio de 114.76 USD o menos, mostrando que el 75% de los productos tiene un precio menor de este y otro 25% de productos tiene un precio superior a 114.76 USD


- **Rating:**
    - **Promedio:** 3, Los productos poseen en promedio un rating de 3 puntos
    - **Rango:**: De 1 a 5, hay productos con 1 punto y otros con muy buenos puntos (hasta 5).
    - **Desviación Estandar:** 1.16 respecto al promedio, significa que los valores de rating varian 1.16 con respecto al valor del promedio 3, aunque representa el 38%, considero que la variabilidad es moderada ya que se trata de rating, pienso que la mayoria de los valores estan cerca de la media pero aun asi existe algo de variabilidad o dispersión.

- **Number_of_Reviews:**
    - **Promedio:** 5014.23, Los productos poseen en promedio alrededor de 5000 reseñas.
    - **Rango:** De 52 a 10000, Hay productos con pocas reseñas(52) y otros con muchas reseñas (hasta 10.000).
    - **Desviación Estandar:** 2855.66 respecto al promedio, podria decirse que es una variabilidad media a alta, ya que representa el 57 % del promedio, esto sugiere que hay productos tanto con pocas como con muchas reseñas, reflejando diversidad en la cantidad de reseñas recibidas.

In [29]:
df.isnull().sum()

Product_Name         0
Brand                0
Category             0
Usage_Frequency      0
Price_USD            0
Rating               0
Number_of_Reviews    0
Product_Size         0
Skin_Type            0
Gender_Target        0
Packaging_Type       0
Main_Ingredient      0
Cruelty_Free         0
Country_of_Origin    0
dtype: int64